### DISCUSSION TITLE
Location filtering with structural variants

### DISCUSSION THREAD

**Discussion by tranthn** [broadinstitute/seqr#3275] [Q&A] [answered]
I am a developer adding the structural variant code to our fork of seqr and was confused by some of the behavior of the location filter when I was searching on some SVs.

I was looking at the latest code and noticed 1 line in the location filter logic that I didn't understand and tracked it down to this original commit: 0d3572a86, specifically this line:

`q |= Q('range', xpos={'lte': xstart}) & Q('range', xstop={'gte': xstop})`

This yields an Elasticsearch query that will return structural variants that have an `xpos < xstart` and `xstop > xstop`.

If I search for something in 1:2000001234-2000005678, it could return hits that have xpos=12000001111 and xstop=2000007777.

I also looked at the latest code, but this behavior seems to be intact, though I saw the corresponding tests in `es_utils_tests.py` did change. Is the above behavior as intended or am I perhaps missing context on my end?

Thank you!

**Comment by hanars** [comment, marked as answer]
This is desired behavior. If you are interested in a region of the genome, we assume you are interested in an SV that contains that interval. For example, if the interval you are searching is a single gene and there is a huge SV that spans many genes including the gene you searched, that SV should be returned

**Reply by tranthn** [reply]
Thank you Hana for the response!

I realized after the fact I may have had a typo in my original question and also didn't clarify the specific cases I was looking at. 

I meant for my above example to be just on chromosome 1 (I accidentally wrote the xstop on chromosome 2). If my range was `xpos=1200001234` and `xstop=1200005678`, it would make sense to return an SV that enveloped that interval, like with `xpos=1200001111` and `xstop=1200007777`.

However, if I had a breakend/BND (or maybe insertion/INS) variant where:
-  the start chromosome is 1 and 'xpos=1200001111`
-  the second chromosome is 18 with `xstop=18200007777`

And I do a search for an interval on chromosome 1, e.g.  `1:200001234-200005678`

Since 1200001111 < 1200001234 AND 18200007777 > 1200005678, that variant would be returned. Is that also expected or should inter-chromosomal translocations be considered another way?

Thank you!

**Reply by hanars** [reply]
We omit breakends from many of our searches so none of our users have ever ran into that edge case, but you are correct that that variant would be incorrectly returned and there is a bug here. I don't have much insight for you as to how you should be considering these events because we don't have great insight into that ourselves.

**Reply by tranthn** [reply]
Thank you, are those breakends omitted via the Annotations chosen in search or using PASS filters, for example? And are translocations (CTXs) and insertions with different end chromosomes treated similarly?

Thank you again for all the insight! I will discuss with my team how we want to handle these cases in our search.

**Reply by hanars** [reply]
They are omitted via chosen annotations. I think for insertions we process them so that use the start and end coordinates for the insertion site and store the location where it is coming from as a detail we display but do not search on. 

Also, if you do have any changes you decide to make for search let me know, we might be interested in integrating them into the main seqr codebase
